### PR TITLE
Add ContextMenuManager

### PR DESCRIPTION
**Pull request by Jarlakxen**

### CODE REVIEW

File: src/main/scala/atom/core/ContextMenuManager.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+   */
+  val created: js.UndefOr[js.Function1[Event, Unit]] = js.undefined,
+  
+  /*
+   *  A function that is called to determine whether to display this item on a
+   *  given context menu deployment.
+   */
+  val shouldDisplay: js.UndefOr[js.Function1[Event, Boolean]] = js.undefined
+) extends js.Object
+
```
Comment (laughedelic) [inline src/main/scala/atom/core/ContextMenuManager.scala]: This misses the `.type` attribute, which can be used to add a separator. One way to solve it is to add this object:

```scala
object ContextMenuSeparator extends ContextMenuItem() {
  final val `type`: String = "separator"
}
```

This has a downside of inheriting all the other undefined attributes. 

Another option is to make a simple ADT:

```scala
sealed trait AnyContextMenuItem extends js.Object

class ContextMenuItem ( 
  // ...
) extends AnyContextMenuItem

object ContextMenuSeparator extends AnyContextMenuItem {
  final val `type`: String = "separator"
}
```

**Review by laughedelic — CHANGES_REQUESTED** [review]
Good 👍  There's just one missing thing in this API.

File: src/main/scala/atom/core/ContextMenuManager.scala (hunk 1 above)
Comment (Jarlakxen) [reply src/main/scala/atom/core/ContextMenuManager.scala]: I was thinking how could I implement that field and I didn't think in using a `object`. Great suggestion!

File: src/main/scala/atom/core/ContextMenuManager.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   *  given context menu deployment.
+   */
+  val shouldDisplay: js.UndefOr[js.Function1[Event, Boolean]] = js.undefined
+) extends AnyContextMenuItem
+
+
+/* Provides a registry for commands that you'd like to appear in the context menu. */
+@js.native
+trait ContextMenuManager extends js.Object {
+  def add(itemsBySelector: js.Dictionary[js.Array[ContextMenuItem]]): Disposable = js.native
```
Comment (laughedelic) [inline src/main/scala/atom/core/ContextMenuManager.scala]: Now this has to be changed to `AnyContextMenuItem` to accept both subtypes

**Review by laughedelic — CHANGES_REQUESTED** [review]

File: src/main/scala/atom/core/ContextMenuManager.scala (hunk 2 above)
Comment (Jarlakxen) [reply src/main/scala/atom/core/ContextMenuManager.scala]: Done!

Comment (laughedelic) [reply src/main/scala/atom/core/ContextMenuManager.scala]: @Jarlakxen Do you have a Github twin @facundo-viale-olx? 😄

Comment (facundo-viale-olx) [reply src/main/scala/atom/core/ContextMenuManager.scala]: Yeap It's really annoying when you have multiple accounts -_-
I used to have just one.

**Pull request merged**